src/services/safeBlock.ts

```diff
@@ -0,0 +1,34 @@
+import config from "config";
+
+import { Pool } from "pg";
+import { Request, Response } from "express";
+
+import { CardanoFrag } from "../Transactions/types";
+
+const safeBlockQuery = `SELECT epoch_no AS "epoch",
+    epoch_slot_no AS "slot",
+    encode(hash, 'hex') as hash,
+    block_no AS height
+FROM BLOCK
+WHERE block_no <= (SELECT MAX(block_no) FROM block) - ($1)::int
+ORDER BY id DESC
+LIMIT 1;
+`;
+
+export const handleSafeBlock = (pool: Pool) => async (req: Request, res: Response) => {
+    const safeBlockDifference = parseInt(config.get("safeBlockDifference"));
+
+    const result = await pool.query(safeBlockQuery, [safeBlockDifference]);
+    if (result.rowCount === 0) throw new Error("no blocks found!");
```

Comment:
This error message and error in general are not strictly correct here. As we are querying a block of a certain required depth it technically might be a valid case when a blockchain does not have a safeblock of required depth yet. I think it will make sense to make the return value compatible with the `/bestblock` endpoint in this case, which returns `ok`, but sets value to `undefined`. We can do same here and then will just handle it on the frontend.

Reply:
I think we should just change the error message here. In the case of `/bestblock`, if it happens that it cannot find the best block the api will actually error out because it doesn't do the `result.rowCount > 0` check. I think there was a confusion here because the `bestblock.ts` doesn't export the request handler, but just an intermediary function used in the actual handler at `src/index.ts`. See my response on your other comment to have more context on that: https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/279#discussion_r740237022

Reply:
Changed to return undefined in case the safe block cannot be found.